src/town_cmd.cpp

```diff
@@ -1263,7 +1330,7 @@ static void GrowTownInTile(TileIndex *tile_ptr, RoadBits cur_rb, DiagDirection t
 		}
 
 		/* Don't walk into water. */
-		if (HasTileWaterGround(house_tile)) return;
+		if (HasTileWaterGround(house_tile) || GrowingOnWateredTile(house_tile)) return;
```

Comment:
If you leave the comment in the previous line, it can be misleading.
As the two checks are independent, and checking logically different thins, it might be better to have your function a separate if line.
Same for the next two occurrences.

Reply:
Thanks, I'm gonna rename it.

I'm currently wondering where exactly in the code I should've placed the checks. I'm worried I could be checking in wrong places, or be missing in some places, or break some logic in the growth algorithm. I'm unsure about the 2x1, 2x2, 1x2 houses, and some of the road checks seem to be of the kind "can we build a road here in the future" which could influence the growth logic somehow.

Need a town growth expert tom come in and to tell where is best to place the checks.